compiler/src/dotty/tools/dotc/transform/TransformByNameApply.scala

```diff
@@ -47,7 +47,7 @@ abstract class TransformByNameApply extends MiniPhase { thisPhase: DenotTransfor
             ref(defn.cbnArg).appliedToType(argType).appliedTo(arg).withSpan(arg.span)
           arg match {
             case Apply(Select(qual, nme.apply), Nil)
-            if qual.tpe.derivesFrom(defn.FunctionClass(0)) && isPureExpr(qual) =>
+            if qual.tpe.derivesFrom(defn.FunctionClass(0)) && (isPureExpr(qual) || qual.symbol.isAllOf(Inline | Param)) =>
```

Comment:
Does it matter? If I were correct, the body of inlined methods is never executed, thus only the TASTY matters.

Reply:
In theory that seems to be the case. But I caught one that came for the `DottyPredef.assert` when compiling dotty itself. It might be a re-bootstrapping issue.

Reply:
I see. It could be some invariants after the typer -- and it's always good to do so.